### PR TITLE
Documentation update for ImageRecordIter

### PR DESCRIPTION
**Pull request by pracheer**
Due to time constraints wasn't able to dive deep into the code to understand more about each of the parameters. Let me know if that is necessary in order to proceed.

### CODE REVIEW

File: src/io/image_iter_common.h [hunk 1]
```diff
@@ -106,9 +106,12 @@ struct ImageRecParserParam : public dmlc::Parameter<ImageRecParserParam> {
   // declare parameters
   DMLC_DECLARE_PARAMETER(ImageRecParserParam) {
     DMLC_DECLARE_FIELD(path_imglist).set_default("")
-        .describe("Path to the image list file");
+        .describe("Path to the image list (.lst) file. Generally created with tools/im2rec.py. "\
+                  "Format (Tab separated): "\
+                  "<Record of index>\t<one or more labels>\t<relative path from root folder>.");
```
Comment (piiswrong) [inline src/io/image_iter_common.h]: index of record?

File: src/io/image_iter_common.h [hunk 2]
```diff
@@ -196,13 +199,13 @@ struct ImageNormalizeParam :  public dmlc::Parameter<ImageNormalizeParam> {
...
   DMLC_DECLARE_PARAMETER(ImageNormalizeParam) {
     DMLC_DECLARE_FIELD(seed).set_default(0)
-        .describe("The Random Seed.");
+        .describe("The random seed.");
     DMLC_DECLARE_FIELD(mirror).set_default(false)
-        .describe("If or not mirror the image.");
+        .describe("Whether to mirror the image or not.");
     DMLC_DECLARE_FIELD(rand_mirror).set_default(false)
-        .describe("If or not randomly the image.");
+        .describe("Randomly chose whether to mirror an image or not.");
```
Comment (piiswrong) [inline src/io/image_iter_common.h]: `If true, 50% of the images will be randomly mirrored (flipped along the horizontal axis)`

**Comment by pracheer** [comment]
@piiswrong Updated as per the feedback.

File: src/io/image_iter_common.h [hunk 3]
```diff
@@ -106,9 +106,16 @@ struct ImageRecParserParam : public dmlc::Parameter<ImageRecParserParam> {
   // declare parameters
   DMLC_DECLARE_PARAMETER(ImageRecParserParam) {
     DMLC_DECLARE_FIELD(path_imglist).set_default("")
-        .describe("Path to the image list file");
+        .describe("Path to the image list (.lst) file. Generally created with tools/im2rec.py. "\
+                  "Format (Tab separated): "\
+<<<<<<< HEAD
```
Comment (piiswrong) [inline src/io/image_iter_common.h]: merge conflicts.

Please rebase to get rid of changes from other people.
```
git fetch upstream
git rebase upstream/master
```

**Comment by pracheer** [comment]
@piiswrong My bad. Updated.

File: src/io/image_iter_common.h [hunk 4]
```diff
@@ -196,13 +199,16 @@ struct ImageNormalizeParam :  public dmlc::Parameter<ImageNormalizeParam> {
...
     DMLC_DECLARE_FIELD(seed).set_default(0)
-        .describe("The Random Seed.");
+        .describe("The random seed.");
     DMLC_DECLARE_FIELD(mirror).set_default(false)
-        .describe("If or not mirror the image.");
+        .describe("Whether to mirror the image or not. If true, images are "\
+                  "flipped along the horizontal axis.");
     DMLC_DECLARE_FIELD(rand_mirror).set_default(false)
-        .describe("If or not randomly the image.");
+        .describe("Randomly chose whether to mirror an image or not. If true,"\
```
Comment (piiswrong) [inline src/io/image_iter_common.h]: Still a little weird. Maybe
Whether to randomly mirror images or not.

Comment (pracheer) [inline src/io/image_iter_common.h]: @piiswrong Yeah that's true. Updated the PR as suggested.

**Pull request merged**